tests/test_qp.py

```diff
@@ -37,48 +36,68 @@ def test_create_qp_no_attr_connected(self):
                             assert qp.qp_state == e.IBV_QPS_RESET, 'RC QP should have been in RESET'
                     except PyverbsRDMAError as ex:
                         if ex.error_code == errno.EOPNOTSUPP:
-                            raise unittest.SkipTest('Create QP with RC attrs is not supported')
+                            raise unittest.SkipTest('Create RC QP without attrs is not supported')
                         raise ex
+
+    def test_create_uc_qp_no_attr(self):
+        """
+        Test UC QP creation via ibv_create_qp without a QPAttr object proivded.
+        """
+        for ctx, attr, _ in self.devices:
+            with PD(ctx) as pd:
+                with CQ(ctx, 100, None, None, 0) as cq:
+                    qia = get_qp_init_attr(cq, attr)
                     qia.qp_type = e.IBV_QPT_UC
                     try:
                         with QP(pd, qia) as qp:
                             assert qp.qp_state == e.IBV_QPS_RESET, 'UC QP should have been in RESET'
                     except PyverbsRDMAError as ex:
                         if ex.error_code == errno.EOPNOTSUPP:
-                            raise unittest.SkipTest('Create QP with UC attrs is not supported')
+                            raise unittest.SkipTest('Create UC QP without attrs is not supported')
                         raise ex
 
+    def test_create_ud_qp_no_attr(self):
+        """
+        Test UD QP creation via ibv_create_qp without a QPAttr object proivded.
+        """
+        for ctx, attr, _ in self.devices:
+            with PD(ctx) as pd:
+                with CQ(ctx, 100, None, None, 0) as cq:
+                    qia = get_qp_init_attr(cq, attr)
+                    qia.qp_type = e.IBV_QPT_UD
+                    try:
+                        with QP(pd, qia) as qp:
+                            assert qp.qp_state == e.IBV_QPS_RESET, 'UD QP should have been in RESET'
+                    except PyverbsRDMAError as ex:
+                        if ex.error_code == errno.EOPNOTSUPP:
+                            raise unittest.SkipTest('Create UD QP without attrs is not supported')
+                        raise ex
 
-    def test_create_qp_no_attr(self):
+    def test_create_raw_qp_no_attr(self):
```

Comment:
you can use the "@requires_root_on_eth" decorator (from utils) for raw_qp tests (in case you're using one device/port)

Reply:
yes, I'll give it a try.